### ISSUE TITLE
Missing license

### ISSUE THREAD

**Issue by nirvdrum** [freenas/iocage-ix-plugins#53]
This project doesn't appear to have a license attached to it, which unfortunately means it's not freely available for use by default. I don't know if you can or event want to impose a license on 3rd party contributions at this time, since each contributor would need to sign off a re-license, but it would be nice if the template files had a license.

**Comment by davidjb** [comment]
Updated my issue in Redmine about contributing guidelines (https://redmine.ixsystems.com/issues/27829) to cover licences as well.  Quite a few FreeNAS repos have the same issue as this one; no clear licence and thus contributing and re-use are ambiguous at present.  Feel free to lend support there since GitHub issues don't [tend to be used](https://github.com/freenas/webui/issues/168).

**Comment by nirvdrum** [comment]
Thanks for the feedback. I filed the same issue on Redmine as well, since I couldn't work out which issue repository is being used:

https://redmine.ixsystems.com/issues/58779

If issues here aren't being used, an admin should just disable them.

**Comment by davidjb** [comment]
Indeed, I think this issue tracker is just one that perhaps got missed from being turned off.  Hopefully the licences can get stored out -- it's definitely a concern of mine as I'd like to contribute (and reuse the code!).

**Comment by miwi-fbsd** [comment]
Hi,

I'll add the License in a while, it is under BSD 2 clause.

**Comment by Ornias1993** [comment]
Okey, I want to explain something:
Arbitrary code isn't copyrightable.

Is this arbitrary? most of this is, yes.
It follows the documented creation guides for these files and only variation is pointing to the right plugin location, this is considered uncreative and arbitrary in many lawsuits before. Hence with or without licence it's basically fair game.

**Comment by nirvdrum** [comment]
@Ornias1993 I've read this a few times and either don't understand some legal term or something was lost in translation. Do you mean "obvious"? If not, could you please point me to something about arbitrary code copyright? I've searched and couldn't find anything, but it may be that it's dependent on jurisdiction.

In any event, the problem with this logic is I need to read the code before I can determine whether it's "arbitrary" or not. If it turns out not to be, I've now tainted things. So, it's far safer just not to look at code without a license since it's not open source software.

**Comment by Ornias1993** [comment]
It has nothing to do with code... But yes: Arbitrary/Obvious...
I tried to explain without legal terms, but what you need to look for is "Originality". 

I think this is a good quote that would lead to a mere config file not passing the Originality treshhold:

> Merely collecting a number of facts and putting them on paper is not sufficient

_https://www.iusmentis.com/copyright/crashcourse/requirements/#:~:text=get%20copyright%20protection.-,Originality,on%20paper%20is%20not%20sufficient._

And no, for the most part that isn't dependent on jurisdiction.

> In any event, the problem with this logic is I need to read the code before I can determine whether it's "arbitrary" or not. If it turns out not to be, I've now tainted things. So, it's far safer just not to look at code without a license since it's not open source software.

Since when do you need a licence to look at something that is published? You don't!
It isn't unlawfull to watch a ripped movie either, it's unlawfull to posses or distribute one.

You can't be sued just because your code looks similar, that has been tried and voided by the courts many times. There is a HIGH burden of proof on the plaintiff to prove you have willingly and significantly copied his copyrighted code, in such a way that it's a violation of copyright.

Simply put:
1. You bother too much
2. This is a repository if install scripts, the chance anything here is copyrightable is minimal at best.
3. If you are so scared about copyright violations, WHY are you posting this? Because you now provided the proof yourself for your hypothetical plaintif to prove you read the code.

**Comment by nirvdrum** [comment]
If it's an open source project, it should have an open source license.

Simply put:

1. You've gone well out of your way to do what? Attack me for asking about an open source license?
2. I guess all software installers are uncopyrightable.
3. I can look at a file listing and determine if there's a license without looking at the code. If you put the license in a sensible place, GitHub will even highlight it for you.

The number of FreeNAS plugins, community or official, is astonishingly small. Rather than attack people looking to contribute, maybe the barriers to entry should be addressed. If this is such a non-issue, applying a license would do no harm and be of trivial effort.

**Comment by jsegaert** [comment]
It looks like all the _unofficial_ plugins are being migrated to Community Plugins Organization.
There is a [license](https://github.com/ix-plugin-hub/documentation/blob/master/LICENSE.md) posted in there.  Please take a look.
